src/nvim/ex_getln.c

```diff
@@ -4308,11 +4306,12 @@ in_history (
       last_i = i;
     }
     history[type][i].hisnum = ++hisnum[type];
-    history[type][i].viminfo = FALSE;
     history[type][i].hisstr = str;
-    return TRUE;
+    history[type][i].timestamp = os_time();
+    history[type][i].additional_elements = NULL;
```

Comment:
And because all lists are arranged in a linked list, so you will not ever have an unreferenced list. I will just add a `list_unref` call without any tests because I have no idea how to test this.